src/modules/jcms_rest/src/Plugin/rest/resource/PromotionalCollectionItemRestResource.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+namespace Drupal\jcms_rest\Plugin\rest\resource;
+
+use Drupal\jcms_rest\Exception\JCMSNotFoundHttpException;
+use Drupal\jcms_rest\Response\JCMSRestResponse;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Provides a resource to get view modes by entity and bundle.
+ *
+ * @RestResource(
+ *   id = "promotional_collection_item_rest_resource",
+ *   label = @Translation("Promotional collection item rest resource"),
+ *   uri_paths = {
+ *     "canonical" = "/promotional-collections/{id}"
+ *   }
+ * )
+ */
+class PromotionalCollectionItemRestResource extends AbstractRestResourceBase {
+
+  /**
+   * Responds to GET requests.
+   *
+   * Returns a list of bundles for specified entity.
+   *
+   * @throws JCMSNotFoundHttpException
+   */
+  public function get(string $id) : JCMSRestResponse {
+    if ($this->checkId($id)) {
+      $query = \Drupal::entityQuery('node')
+        ->condition('changed', \Drupal::time()->getRequestTime(), '<')
```

Comment:
what's the time about? Something related to multiple versions?

Reply:
These conditions could go on single item queries. I can't honestly remember the significance of this condition for lists but it might protect against any date scheduled for publication in the future. Will remove for all item queries in separate PR and refactor here.

Reply:
It's gone.